snuba/datasets/factory.py

```diff
@@ -53,12 +54,19 @@ def get_dataset(name: str) -> Dataset:
 
     try:
         dataset = DATASETS_IMPL[name] = dataset_factories[name]()
+        DATASETS_NAME_LOOKUP[dataset] = name
     except KeyError as error:
         raise InvalidDatasetError(f"dataset {name!r} does not exist") from error
 
     return dataset
 
 
+def get_dataset_name(dataset: Dataset):
+    if dataset not in DATASETS_NAME_LOOKUP:
```

Comment:
I don't think there's any way this can happen in practice, since hopefully we're not instantiating `Dataset` classes outside of `get_dataset`

Reply:
I think we'll want to do this eventually for testing purposes (my pipe dream is that we can separate the Sentry-specific code at some point, so all of the tests that rely on `events` will need to be generalized), but we'll always need some sort of registry, so we can cross that bridge when we come to it (if ever.)

For now, I think your assumption is the right one.